src/storage/s3/storage.c

```diff
@@ -662,65 +647,14 @@ storageS3InfoList(
 
     MEM_CONTEXT_TEMP_BEGIN()
     {
-        StorageS3InfoListData data = {.callback = callback, .callbackData = callbackData};
-        storageS3ListInternal(this, path, false, false, storageS3InfoListCallback, &data);
+        StorageS3InfoListData data = {.level = level, .callback = callback, .callbackData = callbackData};
+        storageS3ListInternal(this, path, param.expression, false, storageS3InfoListCallback, &data);
```

Comment:
I don't see any regExp tests in s3Test (maybe you can point them out?) - is it not possible/practical to have a test there?

Reply:
There are tests in s3Test.c at 935,-940. These are calling storageList() but storageList() calls storageInfoList().

Reply:
Converted these to storageInfoList() tests to make the coverage more obvious.